src/array/empty.js

```diff
@@ -0,0 +1,23 @@
+define(['../lang/isArray'], function(isArray) {
+
+    /**
+     * Array empty
+     */
+
+    function empty(array) {
+        var i, len;
+
+        if(!isArray(array)) {
+            return [];
+        }
+
+        for (i = 0, len = array.length; i < len; i++) {
+            array.pop();
+        }
+
```

Comment:
> The performance point is still valid, but might be negligible.

I think it's a micro-opt. It seems counter intuitive too but so does a lot of JS perfy things ;)